lib/modules/manager/github-actions/community.ts

```diff
@@ -0,0 +1,144 @@
+import { z } from 'zod';
+
+import type { SkipReason, StageName } from '../../../types';
+import { escapeRegExp, regEx } from '../../../util/regex';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { NpmDatasource } from '../../datasource/npm';
+import { PypiDatasource } from '../../datasource/pypi';
+import * as condaVersioning from '../../versioning/conda';
+import * as npmVersioning from '../../versioning/npm';
+import * as pep440versioning from '../../versioning/pep440';
+import type { PackageDependency } from '../types';
+
+function matchAction(action: string): z.Schema {
+  return z
+    .string()
+    .regex(regEx(`(?:https?://[^/]+/)?${escapeRegExp(action)}(@.+)?$`));
+}
+
+const setupUVSchema = z
+  .object({
+    // https://github.com/astral-sh/setup-uv
+    uses: matchAction('astral-sh/setup-uv'),
+    with: z.object({ version: z.string().optional() }),
+  })
+  .transform(({ with: val }): PackageDependency => {
+    let skipStage: StageName | undefined;
+    let skipReason: SkipReason | undefined;
+
+    if (!val.version) {
+      skipStage = 'extract';
+      skipReason = 'unspecified-version';
+    }
+
+    return {
+      datasource: GithubReleasesDatasource.id,
+      depName: 'astral-sh/uv',
+      versioning: npmVersioning.id,
+      packageName: 'astral-sh/uv',
+      skipReason,
+      skipStage,
```

Comment:
```suggestion```suggestion
      ...(skipStage && { skipStage }),
      ...(skipReason && { skipReason }),
```
```